File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\DBAL\Types\Types;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

#[ORM\Entity]
class Evenement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ? int $idEvenement=null;

    #[ORM\Column(length : 255)]
    private ? string $nom=null;

    #[ORM\Column(length : 255)]
    private ? string $descriptif=null;
    

    #[ORM\Column(length : 255)]
    private ? string $image=null;
    

    #[ORM\Column(length : 255)]
    private ? int $likes=null;

    #[ORM\Column(length : 255)]
    private ? int $nbParticipants=null;

    
    #[ORM\Column(type: "date")]
    #[Assert\NotBlank(message: "start date is required")]
    private $dateDebut;

    #[ORM\Column(type: "date")]
    #[Assert\NotBlank(message: "expiration date is required")]
    #[Assert\GreaterThan(propertyPath:"dateDebut", message:"expiration date must be greater than start date")]
    private $dateFin;

   
    #[ORM\ManyToOne(targetEntity: Categorie::class)]
    #[ORM\JoinColumn(name: "id_categorie", referencedColumnName: "id_categorie")]
    private ?Categorie $idCategorie = null;

    #[ORM\OneToMany(targetEntity: Participation::class, mappedBy: "evenement", orphanRemoval: true)]
    private ?Collection $participations = null;

    public function getIdEvenement(): ?int
    {
        return $this->idEvenement;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getDescriptif(): ?string
    {
        return $this->descriptif;
    }

    public function setDescriptif(string $descriptif): self
    {
        $this->descriptif = $descriptif;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getLikes(): ?int
    {
        return $this->likes;
    }

    public function setLikes(int $likes): self
    {
        $this->likes = $likes;

        return $this;
    }

    public function getNbParticipants(): ?int
    {
        return $this->nbParticipants;
    }

    public function setNbParticipants(int $nbParticipants): self
    {
        $this->nbParticipants = $nbParticipants;

        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->dateDebut;
    }

    public function setDateDebut(\DateTimeInterface $dateDebut): self
    {
        $this->dateDebut = $dateDebut;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(\DateTimeInterface $dateFin): self
    {
        $this->dateFin = $dateFin;

        return $this;
    }

    public function getIdCategorie(): ?Categorie
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(?Categorie $idCategorie): self
    {
        $this->idCategorie = $idCategorie;

        return $this;
    }
/**
     * @return Collection<int, Participation>
     */
    public function getParticipations(): Collection
    {
        return $this->participations;
    }

    public function addParticipation(Participation $participation): self
    {
        if (!$this->participations->contains($participation)) {
            $this->participations[] = $participation;
            $participation->setEvenement($this);
        }

        return $this;
    }

    public function removeParticipation(Participation $participation): self
    {
        if ($this->participations->removeElement($participation)) {
            // set the owning side to null (unless already changed)
            if ($participation->getEvenement() === $this) {
                $participation->setEvenement(null);
            }
        }

        return $this;
    }



}

File: AnasHammou/src/tn/gestion/evenement/enitite/Evenement.java
package tn.gestion.evenement.enitite;


public class Evenement {
    
    private int idEvenement;
    private String nom;
    private String descriptif;
    private String image;
    private int likes;
    private String date_debut;
    private String date_fin;
    private int nbParticipants;
    private Categorie idCategorie;

    public Evenement() {
    }

    public Evenement(int idEvenement, String nom, String descriptif, String image, int likes, String date_debut, String date_fin, int nbParticipants, Categorie idCategorie) {
        this.idEvenement = idEvenement;
        this.nom = nom;
        this.descriptif = descriptif;
        this.image = image;
        this.likes = likes;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.nbParticipants = nbParticipants;
        this.idCategorie = idCategorie;
    }

    public Evenement(String nom, String descriptif, String image, int likes, String date_debut, String date_fin, int nbParticipants, Categorie idCategorie) {
        this.nom = nom;
        this.descriptif = descriptif;
        this.image = image;
        this.likes = likes;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.nbParticipants = nbParticipants;
        this.idCategorie = idCategorie;
    }

    public int getIdEvenement() {
        return idEvenement;
    }

    public void setIdEvenement(int idEvenement) {
        this.idEvenement = idEvenement;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescriptif() {
        return descriptif;
    }

    public void setDescriptif(String descriptif) {
        this.descriptif = descriptif;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getLikes() {
        return likes;
    }

    public void setLikes(int likes) {
        this.likes = likes;
    }

    public String getDate_debut() {
        return date_debut;
    }

    public void setDate_debut(String date_debut) {
        this.date_debut = date_debut;
    }

    public String getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(String date_fin) {
        this.date_fin = date_fin;
    }

    public int getNbParticipants() {
        return nbParticipants;
    }

    public void setNbParticipants(int nbParticipants) {
        this.nbParticipants = nbParticipants;
    }

    public Categorie getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(Categorie idCategorie) {
        this.idCategorie = idCategorie;
    }

    @Override
    public String toString() {
        return "Evenement{" + "idEvenement=" + idEvenement + ", nom=" + nom + ", descriptif=" + descriptif + ", image=" + image + ", likes=" + likes + ", date_debut=" + date_debut + ", date_fin=" + date_fin + ", nbParticipants=" + nbParticipants + ", idCategorie=" + idCategorie + '}';
    }
    
    
}
